src/azul/service/repository_controller.py

```diff
@@ -60,12 +133,31 @@ def download_file(self,
                                     f'version {file_version!r} in catalog {catalog!r}')
             file_version = file['version']
             drs_path = file['drs_path']
+            file_size = file['size']
             if file_name is None:
                 file_name = file['name']
         else:
             assert file_version is not None
             assert file_name is not None
 
+        # Due to https://github.com/curl/curl/issues/6740 causing curl to error
+        # when trying to resume a previously completed file download, we check
+        # for a range request starting at the end of the file and instead of
+        # a returning a 416 (Range Not Satisfiable) as specified in RFC7233
+        # https://tools.ietf.org/html/rfc7233#section-4.4 we return a 206
+        # (Partial Content) with an empty body.
+        range_specifier = headers.get('range')
+        if range_specifier is not None and file_size:
+            requested_range = self._parse_range_request_header(range_specifier)
+            if requested_range == [(file_size, None)]:
+                last_byte_pos = file_size - 1
+                content_range = f'bytes {last_byte_pos}-{last_byte_pos}/{file_size}'
```

Comment:
This is a range of exactly one byte which contradicts a content-length of 0. We are already violating the standard so we shouldn't make the violation more "elaborate" than absolutely necessary. You said you didn't observe any differences whether this is included or not, so I'm going to lean towards leaving the Content-Range response header out.